EDU-112-86: Section override (Part I - Edited files) (#1111)
vtexdocs/dev-portal-content: docs/faststore/docs/customization/building-sections/overrides/component-props.mdx

---
title: 'Overriding native component's props'
---

Overriding props allows you to customize a component's behavior while preserving its native integration. This approach is useful when you want to customize specific aspects of a native component, for example, an icon or a button label. 

For this guide, we'll use the [Buy Button](https://developers.vtex.com/docs/guides/faststore/molecules-buy-button) component to make the customization. 

Here's how the initial component looks with the `size` prop set as `regular` in the Product Details Page (PDP):

![overrides-PDP-buy-button-initial](https://vtexhelp.vtexassets.com/assets/docs/src/overrides-pdp-buy-button-initial___a4b7ae9d88ed888da5649b2b6cc1eecd.png)

## Before you begin

Make sure your `@faststore/core` package has the 3.x version or above. If you need to update
it, refer to this [release note](https://github.com/vtex/faststore/releases/tag/v3.0.0).

## Step by step
### Step 1 - Setting up the component file

> 1. After choosing a native section to be customized from the [list of available native sections](/LINK), open your FastStore project in any code editor of your preference.

Author: ```suggestion
1. After choosing a native section to be customized from the [list of available native sections](https://developers.vtex.com/docs/guides/faststore/building-sections-list-of-native-sections), open your FastStore project in any code editor of your preference.
```

    ```tsx ProductDetailsWithCustomButton.tsx mark=5,6,7
    import { ProductDetailsSection, getOverriddenSection } from '@faststore/core'
    
    const ProductDetailsWithCustomButton = getOverriddenSection({
    Section: ProductDetailsSection,
    components: {
        BuyButton: { props: { size: "small", iconPosition: "right" } } 
    },
    });
    
    export default ProductDetailsWithCustomButton;
    ```

2. In the `components` folder, create the `index.tsx` file and add the following:

    ```tsx index.tsx
    import ProductDetailsWithCustomButton from "./sections/ProductDetailsWithCustomButton";

    export default { ProductDetailsWithCustomButton };

    ```

This code overrides the `BuyButton` props `size` and `iconPosition`. You'll see a smaller button than the initial one, and the cart icon positioned on the right side.

### Step 3 - Adding the component to the CMS

To integrate the section into the [Headless CMS](https://developers.vtex.com/docs/guides/faststore/headless-cms-overview), follow the steps below:

> 1. Create a folder named `cms` in the FastStore root directory.
> 
> 2. Inside `cms`, create the `faststore` folder.
> 
> 3. Within the `cms/faststore` folder, create the `sections.json` file.
> 
> 4. Add the schema of the new section to the `sections.json` file, named `ProductDetailsWithCustomButton`, to the `sections.json` file:

Reviewer: ```suggestion
1. Create a folder named `cms` in the FastStore root directory.
2. Inside `cms`, create the `faststore` folder.
3. Within the `cms/faststore` folder, create the `sections.json` file.
4. Add the schema of the new section to the `sections.json` file, named `ProductDetailsWithCustomButton`, to the `sections.json` file:
```

                    "to": { "title": "URL", "type": "string" }
                }
                },
                "shippingOptionsTableTitle": {
                "title": "Shipping Options Table Header",
                "type": "string"
                }
            }
            },
            "productDescription": {
            "title": "Product Description",
            "type": "object",
            "properties": {
                "initiallyExpanded": {
                "type": "string",
                "title": "Initially Expanded?",
                "enumNames": ["First", "All", "None"],
                "enum": ["first", "all", "none"]
                },
                "displayDescription": {
                "title": "Should display description?",
                "type": "boolean",
                "default": true
                },
                "title": {
                "title": "Description section title",
                "type": "string",
                "default": "Description"
                }
            }
            }
        }
        }
    }
    ]
    ```

This schema Adds the new `Product Details with custom button` section to the Headless CMS.

> 5. Open a new terminal window and log in to your vtex account by running `vtex login {accountName}`.
> 
> 6. run the following command to sync your changes with the Headless CMS:
> 
>     ```bash
>     yarn cms-sync
>     ```
> 
> 7. Access the VTEX Admin and proceed to **Storefront > Headless CMS**.
> 
> 8. Click on the page you desire to add the new section.
> 
> 9. Click the add button (`+`) and add the new section and complete values from the sections fields.
> 
> 10. Click `Save`.

Reviewer: ```suggestion
5. Open a new terminal window and log in to your vtex account by running `vtex login {accountName}`.
6. run the following command to sync your changes with the Headless CMS:

    ```bash
    yarn cms-sync
    ```

7. Access the VTEX Admin and proceed to **Storefront > Headless CMS**.
8. Click on the page you desire to add the new section.
9. Click the add button (`+`) and add the new section and complete values from the sections fields.
10. Click `Save`.
```
